Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2198 @@
+                    viewableCenter.z = OcclusionPositionPadding.z;
+                    break;
+
+                case ScrollDirectionType.LeftAndRight:
+
+                    //Same as above for L <-> R
+                    clippingBounds.size = new Vector3(clippingBounds.size.x * ViewableArea, clippingBounds.size.y * Tiers, clippingBounds.size.z);
+                    clipBox.transform.localScale = VectorExtensions.ScaleFromBounds(new Bounds(Vector3.zero, Vector3.one), clippingBounds) + OcclusionScalePadding;
+
+                    //Same as above for L <-> R
+                    viewableCenter.x = (clipBox.transform.localScale.x * 0.5f) - (OcclusionScalePadding.x * 0.5f) + OcclusionPositionPadding.x;
+                    viewableCenter.y = ((clipBox.transform.localScale.y * 0.5f) - (OcclusionScalePadding.y * 0.5f) + OcclusionPositionPadding.y) * -1.0f;
+                    viewableCenter.z = OcclusionPositionPadding.z;
+                    break;
+            }
+
+            //Apply our new values
+            clipBox.transform.localPosition = viewableCenter;
+
+            //add our objects to the clippingBox queue
+            AddAllItemsToClippingObject();
+        }
+
+        #endregion ObjectCollection methods
+
+        #region Monobehavior Implementation
+
+        private void OnEnable()
+        {
+            //Register for global input events
+            MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out IMixedRealityInputSystem inputSys);
+
+            if (inputSys != null)
+            {
+                inputSys.RegisterHandler<IMixedRealityTouchHandler>(this);
+                inputSys.RegisterHandler<IMixedRealityPointerHandler>(this);
+            }
+
+            if (useOnPreRender)
+            {
+                if (clippingObject == null || clipBox == null)
+                {
+                    UpdateCollection();
+                }
+
+                clipBox.UseOnPreRender = true;
+
+                //Subscribe to the preRender callback on the main camera so we can intercept it and make sure we catch
+                //any dynamically created children in our list
+                cameraMethods = CameraCache.Main.gameObject.EnsureComponent<CameraEventRouter>();
+                cameraMethods.OnCameraPreRender += OnCameraPreRender;
+            }
+        }
+
+        private void Start()
+        {
+            if (setUpAtRuntime)
+            {
+                UpdateCollection();
+            }
+
+            if (useOnPreRender && clipBox != null)
+            {
+                clipBox.UseOnPreRender = true;
+
+                //Subscribe to the preRender callback on the main camera so we can intercept it and make sure we catch
+                //any dynamically created children in our list
+                cameraMethods = CameraCache.Main.gameObject.EnsureComponent<CameraEventRouter>();
```

Comment:
woah, again.

Reply:
I tweaked this a bit, but there are a bunch of scenarios where you can have UseOnPreRender enabled but these components wouldn't get set up properly.

1. I have a scrollingObjectCollection set up at editor time, (uses OnEnable/OnDisable).
2. I have a dynamic scrollingObjectCollection instantiated from code with UseOnPreRender configured, it exists in the scene, but UpdateCollection() is called at a later time. (uses UpdateCollection/OnDisable).
3. At run time, at a later time, I decide I need UseOnPreRender (uses the Setter/OnDisable).

I'm sure theres a better way to do this, maybe put this in the Getter as well as the Setter and then always reference the property?